source/rust_verify/src/driver.rs

```diff
@@ -134,6 +134,21 @@ pub(crate) fn run_with_erase_macro_compile(
     let mut callbacks =
         CompilerCallbacksEraseMacro { lifetimes_only: !compile, test_capture_output };
     rustc_args.extend(["--cfg", "verus_macro_erase_ghost"].map(|s| s.to_string()));
+    let allow = &[
+        "unused_imports",
+        "unused_variables",
+        "unreachable_patterns",
+        "unused_parens",
+        "unused_braces",
+        "dead_code",
+        "unreachable_code",
+        "unused_mut",
+        "unused_labels",
+        "unused_attributes",
+    ];
```

Comment:
Are these something we think we'd need long-term or are they temporary?

Reply:
I think this is long term, at least as long as we're running `rustc` twice (once to verify and once to compile).  The meaningful warnings are generated only on the first run of `rustc`.  The second run, for compiling erased code, generates spurious warnings about the erased code.

At least, this is the way it appears to work for some warnings, like `unreachable_code` and `unused_imports`.  Some other warnings, unfortunately, currently don't get emitted until very late (they appear only when we use `--compile`), so I don't know how to retain these without mixing them in with the spurious warnings from erasure.  Maybe there's some way to prod rustc to generate them earlier?

If in the future we have support for erasing ghost code directly in rust, with only one run of `rustc`, then this issue goes away.

Reply:
Understood, thanks.

> Maybe there's some way to prod rustc to generate them earlier?

We may be able to override rust's queries (or run them earlier); it may be easier to do once we switch to a pre-built rustc with a custom driver.